documents/coredocument.go

```diff
@@ -249,7 +239,7 @@ func (cd *CoreDocument) CreateProofs(docType string, dataTree *proofs.DocumentTr
 	if err != nil {
 		return nil, errors.New("failed to generate core document tree: %v", err)
 	}
-	srHash, err := cd.GetSigningRootHash()
+	srHash, err := cd.GetSigningRootProofHash(docType, dataTree.RootHash())
```

Comment:
can we rename this to `CalculateSigningRoot`? The should be consistent with remaining root calculations

Reply:
Actually, we dont need this. We need to replace this function call with `CalculateSingatureRoot`.

Reply:
it just leaks info about the tree to outside, I would keep it the way it is, so that we can change/evolve the implementations as we need.